File: frontend/src/types/feed.ts
// Feed types — matching OpenAPI schemas/feeds.yaml

export interface Feed {
  id: string
  team_id: string
  name: string
  description?: string | null
  created_by_user_id: string
  created_at: string
  updated_at: string
}

export interface FeedItem {
  id: string
  team_id: string
  feed_id: string
  project_id?: string | null
  title: string
  content: string
  excerpt: string
  ai_assistant_name: string
  posted_by_user_id: string
  archived_at?: string | null
  posted_at: string
  reply_count?: number
}

export interface FeedItemReply {
  id: string
  team_id: string
  feed_item_id: string
  content: string
  posted_by_user_id: string
  ai_assistant_name?: string | null
  posted_at: string
}

export interface CreateFeedItemReplyRequest {
  content: string
  ai_assistant_name?: string
}

export interface FeedItemReplyListResponse {
  replies: FeedItemReply[]
  total_count: number
  page: number
  per_page: number
  total_pages: number
}

export interface CreateFeedRequest {
  name: string
  description?: string
}

export interface UpdateFeedRequest {
  name?: string
  description?: string
}

export interface CreateFeedItemRequest {
  title: string
  content: string
  ai_assistant_name: string
  project_id?: string
}

export interface FeedListResponse {
  feeds: Feed[]
  total_count: number
  page: number
  per_page: number
  total_pages: number
}

export interface FeedItemListResponse {
  items: FeedItem[]
  total_count: number
  page: number
  per_page: number
  total_pages: number
}

export interface FeedFilters {
  search?: string
  page?: number
  limit?: number
}

export interface FeedItemFilters {
  feed_id?: string
  project_id?: string
  ai_assistant_name?: string
  archived?: 'true' | 'false' | 'all'
  search?: string
  page?: number
  limit?: number
}

File: backend/internal/models/feed.go
package models

import "time"

// FeedItemReply represents a threaded reply to a feed item
type FeedItemReply struct {
	ID              string    `json:"id"                         db:"id"`
	TeamID          string    `json:"team_id"                    db:"team_id"`
	FeedItemID      string    `json:"feed_item_id"               db:"feed_item_id"`
	Content         string    `json:"content"                    db:"content"`
	PostedByUserID  string    `json:"posted_by_user_id"          db:"posted_by_user_id"`
	AIAssistantName *string   `json:"ai_assistant_name,omitempty" db:"ai_assistant_name"`
	PostedAt        time.Time `json:"posted_at"                  db:"posted_at"`
}

// CreateFeedItemReplyRequest is the request body for posting a reply to a feed item
type CreateFeedItemReplyRequest struct {
	Content         string  `json:"content"                    validate:"required,min=1,max=10000"`
	AIAssistantName *string `json:"ai_assistant_name,omitempty" validate:"omitempty,max=30"`
}

// FeedItemReplyListResponse is the paginated response for listing feed item replies
type FeedItemReplyListResponse struct {
	Replies    JSONArray[FeedItemReply] `json:"replies"`
	TotalCount int                      `json:"total_count"`
	Page       int                      `json:"page"`
	PerPage    int                      `json:"per_page"`
	TotalPages int                      `json:"total_pages"`
}

// Feed represents a team-scoped AI feed channel
type Feed struct {
	ID              string    `json:"id" db:"id"`
	TeamID          string    `json:"team_id" db:"team_id"`
	Name            string    `json:"name" db:"name"`
	Description     *string   `json:"description,omitempty" db:"description"`
	CreatedByUserID string    `json:"created_by_user_id" db:"created_by_user_id"`
	CreatedAt       time.Time `json:"created_at" db:"created_at"`
	UpdatedAt       time.Time `json:"updated_at" db:"updated_at"`
}

// FeedItem represents a single item posted to a feed
type FeedItem struct {
	ID              string     `json:"id" db:"id"`
	TeamID          string     `json:"team_id" db:"team_id"`
	FeedID          string     `json:"feed_id" db:"feed_id"`
	ProjectID       *string    `json:"project_id,omitempty" db:"project_id"`
	Title           string     `json:"title" db:"title"`
	Content         string     `json:"content" db:"content"`
	Excerpt         string     `json:"excerpt" db:"excerpt"`
	AIAssistantName string     `json:"ai_assistant_name" db:"ai_assistant_name"`
	PostedByUserID  string     `json:"posted_by_user_id" db:"posted_by_user_id"`
	ArchivedAt      *time.Time `json:"archived_at,omitempty" db:"archived_at"`
	PostedAt        time.Time  `json:"posted_at" db:"posted_at"`
	ReplyCount      int        `json:"reply_count" db:"-"`
}

// CreateFeedRequest is the request body for creating a new feed
type CreateFeedRequest struct {
	Name        string  `json:"name" validate:"required,min=1,max=255"`
	Description *string `json:"description,omitempty" validate:"omitempty,max=1000"`
}

// UpdateFeedRequest is the request body for updating an existing feed
type UpdateFeedRequest struct {
	Name        *string `json:"name,omitempty" validate:"omitempty,min=1,max=255"`
	Description *string `json:"description,omitempty" validate:"omitempty,max=1000"`
}

// CreateFeedItemRequest is the request body for posting an item to a feed
type CreateFeedItemRequest struct {
	Title           string  `json:"title" validate:"required,min=1,max=255"`
	Content         string  `json:"content" validate:"required,min=1"`
	AIAssistantName string  `json:"ai_assistant_name" validate:"required,min=1,max=30"`
	ProjectID       *string `json:"project_id,omitempty" validate:"omitempty,uuid"`
}

// FeedListResponse is the paginated response for listing feeds
type FeedListResponse struct {
	Feeds      JSONArray[Feed] `json:"feeds"`
	TotalCount int             `json:"total_count"`
	Page       int             `json:"page"`
	PerPage    int             `json:"per_page"`
	TotalPages int             `json:"total_pages"`
}

// FeedItemListResponse is the paginated response for listing feed items
type FeedItemListResponse struct {
	Items      JSONArray[FeedItem] `json:"items"`
	TotalCount int                 `json:"total_count"`
	Page       int                 `json:"page"`
	PerPage    int                 `json:"per_page"`
	TotalPages int                 `json:"total_pages"`
}

// FeedWithLastPost is a Feed enriched with the timestamp of the most recent post.
// It is returned by the MCP list-feeds tool and does NOT alter the REST API response shape.
type FeedWithLastPost struct {
	ID              string     `json:"id"`
	TeamID          string     `json:"team_id"`
	Name            string     `json:"name"`
	Description     *string    `json:"description,omitempty"`
	CreatedByUserID string     `json:"created_by_user_id"`
	CreatedAt       time.Time  `json:"created_at"`
	UpdatedAt       time.Time  `json:"updated_at"`
	LastPostAt      *time.Time `json:"last_post_at"`
}

// MCPFeedListResponse is the response returned by the vibexp_io_list_feeds MCP tool.
type MCPFeedListResponse struct {
	Feeds []FeedWithLastPost `json:"feeds"`
}
